akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -525,4 +536,13 @@ object PatternsCS {
    */
   def after[T](duration: FiniteDuration, scheduler: Scheduler, context: ExecutionContext, value: CompletionStage[T]): CompletionStage[T] =
     afterCompletionStage(duration, scheduler)(value)(context)
+
+  /**
+   * Returns an internally retrying [[java.util.concurrent.CompletionStage]]
+   * The first attempt will be made immediately, and each subsequent attempt will be made after 'delay'.
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * If attempts are exhausted the returned completion stage is simply the result of invoking attempt
+   */
+  def retry[T](attempt: Callable[CompletionStage[T]], attempts: Int, delay: FiniteDuration, scheduler: Scheduler, ec: ExecutionContext): CompletionStage[T] =
+    scalaRetry(() => attempt.call().toScala, attempts, delay, scheduler)(ec).toJava
```

Comment:
Not sure I understand what you wrote regarding returning a new attempt, could you elaborate please?